lua/entities/gmod_wire_expression2/core/string.lua

```diff
@@ -337,14 +337,23 @@ end)
 
 /******************************************************************************/
 
+local function checkregex(data, pattern)
+	local n = 0 for i in string.gmatch(string.gsub(pattern, "%%.", ""), "[%+%-%*]") do n = n + 1 end
+	return n==0 or (n==1 and #data<15000) or (n==2 and #data<900)
+end
 
 local sub = string.sub
 local gsub = string.gsub
 local find = string.find
 
 --- Returns the 1st occurrence of the string <pattern>, returns 0 if not found. Prints malformed string errors to the chat area.
 e2function number string:findRE(string pattern)
-	local OK, Ret = pcall(string.find, this, pattern)
+	local OK, Ret
+	if checkregex(this, pattern) then
+		OK, Ret = pcall(string.find, this, pattern)
+	else
+		OK, Ret = false, "regex is too complex!"
```

Comment:
how about some slightly more detailed feedback, such as 'regex too complex' (in case of n > 2), and 'regex of complexity n can only treat strings of x length' or something.